### PR TITLE
1 cleaning

### LINKED ISSUE #1: data cleaning

### PR DESCRIPTION
**Pull request by ElenaMariani**
Closes #1 

To review this PR please follow the instruction in the main readme file that show how to set up the requirements and run the full pipeline.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Simulating Calorie Reduction Values to Meet Obesity Reduction Targets in Scotland
+
+## Requirements
+
+This project was built with R version 4.2.1 (2022-06-23) -- "Funny-Looking Kid" and RStudio 2022.07.2+576 "Spotted Wakerobin" Release (e7373ef832b49b2a9b88162cfe7eac5f22c40b34, 2022-09-06) for macOS Mozilla/5.0 (Macintosh; Intel Mac OS X 12_6\_0) AppleWebKit/537.36 (KHTML, like Gecko) QtWebEngine/5.12.10 Chrome/69.0.3497.128 Safari/537.36
+
+The list of packages needed to run the project is automatically installed by opening and running the script `requirements.R` which automatically checks whether the packages needed for this analysis are installed and if they are not it installs them. This repo relies on the `bw` package which is not currently available on CRAN. To install this package, run the following code.
+
```
Comment (izzyStewart) [inline README.md:8]: Is there a reason why the below code can't be added to the requirements file?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+1.  Follow this [link](https://beta.ukdataservice.ac.uk/datacatalogue/series/series?id=2000047) to access the Scottish Health Survey series
+2.  Click on the Access data button
+3.  On the next page click on Add to account
+4.  Head to your account and add all the datasets to a project
+5.  Head to your project, click on Actions, and then click on Download from the drop down menu
+6.  Select the TAB option and then Download selected
+7.  A zip folder will be downloaded: extract the files and head to the tab folder: the data files needed for this project are the individual level files (with suffix `i`)
+8.  Save the files to `inputs\data`
```
Comment (izzyStewart) [inline README.md:27]: Maybe not an issue but noticed there is no `inputs/data` folder. Maybe just make it explicit that you need to create them.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+-   In the `data_processing` folder open and run the file `read_clean_save_scotland.R` (further info in the README of this folder)
+-   In the `pipeline` folder run the files in this order:
+    -   `equating.R`
+
+    -   `run_hall.R`
+
+    -   `extrapolate.R`
+
+All outputs are saved in the `outputs` folder (`data`, `reports`, and `figures`).
```
Comment (izzyStewart) [inline README.md:43]: There doesn't seem to be an outputs folder? Or is this created on running the code? Any reason why the outputs aren't pushed to the remote?

File: data_processing/read_clean_save_scotland.R [hunk 4]
```diff
@@ -0,0 +1,331 @@
+rm(list = ls())
```
Comment (izzyStewart) [inline data_processing/read_clean_save_scotland.R:1]: Running this file gave me the below error message:

```
Error: Cannot open file for writing:
* '/Users/isabel.stewart/ahl_scotland_modelling/outputs/data/shs_2019_adult_clean.csv'
```

I added the folders `outputs/data` and it ran fine

File: pipeline/equating.R [hunk 5]
```diff
@@ -0,0 +1,74 @@
+library(Hmisc)
```
Comment (izzyStewart) [inline pipeline/equating.R:1]: Similar issue with the data processing file where it initially errors as I don't have a the `outputs/reports` folder. Added it and it ran fine.

File: pipeline/run_hall.R [hunk 6]
```diff
@@ -0,0 +1,71 @@
+require("bw")
```
Comment (izzyStewart) [inline pipeline/run_hall.R:1]: Could be worth flagging the time it takes to run this script just so they know its not frozen ect.

File: README.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
+# Simulating Calorie Reduction Values to Meet Obesity Reduction Targets in Scotland
```
Comment (izzyStewart) [inline README.md:1]: This is really clear / easy to follow. I wonder if you might want to include some of the figures in the readme and briefly talk around them? Nothing too detailed but could be nice to see what some of the outputs look like.

File: pipeline/README.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
+# Pipeline
```
Comment (izzyStewart) [inline pipeline/README.md:1]: Is there a reason why these files need to be run seperately? As in would there be a need to run one but not the others? If not I would have one script at the top that reads the .R files in as modules and runs them in one go.

File: utils/README.md [hunk 9]
```diff
@@ -0,0 +1,3 @@
+# Utils
+
```
Comment (izzyStewart) [inline utils/README.md:2]: I wonder if having a project directory diagram in the main readme might be helpful for the reader? As this folders not referenced in the 'how to run the code' section. We did something similar for the CCI project example here: https://github.com/nestauk/cci_cameroon (see 'directory structure')

File: pipeline/extrapolate.R [hunk 10]
```diff
@@ -0,0 +1,775 @@
+rm(list = ls())
+
```
Comment (izzyStewart) [inline pipeline/extrapolate.R:2]: Running this file gave the error: 

`Error: '/Users/isabel.stewart/ahl_scotland_modelling/outputs/reports/obesity_change.csv' does not exist.`

Looking in the reports folder I can't see it (I'm guessing it should have been created from `run_hall.R`?

**Review by izzyStewart — COMMENTED** [review]
Ran fine until I got to `extrapolate.R` which gave the below error. I have left some other comments mainly about some documentation suggestions and creating folders ahead of running the files.

`Error: '/Users/isabel.stewart/ahl_scotland_modelling/outputs/reports/obesity_change.csv' does not exist.`

File: README.md (hunk 1 above)
Comment (ElenaMariani) [reply README.md:8]: No reason! Added to requirements, thanks

File: README.md (hunk 2 above)
Comment (ElenaMariani) [reply README.md:27]: Edited `.gitignore` thanks

File: README.md (hunk 3 above)
Comment (ElenaMariani) [reply README.md:43]: Edited `.gitignore`.

File: data_processing/read_clean_save_scotland.R (hunk 4 above)
Comment (ElenaMariani) [reply data_processing/read_clean_save_scotland.R:1]: Edited `.gitignore`

File: pipeline/equating.R (hunk 5 above)
Comment (ElenaMariani) [reply pipeline/equating.R:1]: Edited `.gitignore`.

File: pipeline/run_hall.R (hunk 6 above)
Comment (ElenaMariani) [reply pipeline/run_hall.R:1]: Done

File: README.md (hunk 7 above)
Comment (ElenaMariani) [reply README.md:1]: Thanks - will do once findings finalised

File: pipeline/README.md (hunk 8 above)
Comment (ElenaMariani) [reply pipeline/README.md:1]: Good suggestion - implemented it

File: pipeline/extrapolate.R (hunk 10 above)
Comment (ElenaMariani) [reply pipeline/extrapolate.R:2]: I realised I had created it as part of a notebook that I didn't push - created a new script in pipeline that creates it

**Pull request merged**